composition/cache_loader.go

```diff
@@ -30,7 +33,18 @@ func (loader *CachingContentLoader) Load(fd *FetchDefinition) (Content, error) {
 	c, err := loader.load(fd)
 	if err == nil {
 		if fd.IsCachable(c.HttpStatusCode(), c.HttpHeader()) {
-			loader.cache.Set(hash, fd.URL, c.MemorySize(), c)
+			if c.Reader() != nil {
+				var streamBytes []byte
+				streamBytes, err = ioutil.ReadAll(c.Reader())
+				cw := &ContentWrapper{
+					Content:     c,
+					streamBytes: streamBytes,
+				}
+				loader.cache.Set(hash, fd.URL, c.MemorySize(), cw)
+				c.SetReader(cw.Reader())
```

Comment:
Why do you need to set the reader here?
I think it would be simpler to also return the 'cw' on this initial call.